app/controllers/credentials_controller.rb

```diff
@@ -35,10 +35,16 @@ class CredentialsController < ApplicationController
   # This method requires a PUT request. The new password is in the request body.
   def update_password
     password = request.body.read
-    
-    @role.credentials.password = password
-    @role.credentials.save
-    head 204
+
+    begin
+      Util::Password.validate_complexity password
```

Comment:
If this is occurring in `password=`, we should not be duplicating it here.

Reply:
Although, Jason's suggestion to move this into the ActiveModel validation is even better.